app/src/test/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/RecordingControllerTest.kt

```diff
@@ -0,0 +1,66 @@
+package co.netguru.baby.monitor.client.feature.voiceAnalysis
+
+import co.netguru.baby.monitor.RxSchedulersOverrideRule
+import co.netguru.baby.monitor.client.feature.machinelearning.MachineLearning
+import co.netguru.baby.monitor.client.feature.noisedetection.NoiseDetector
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Observable
+import org.junit.Rule
+import org.junit.Test
+
+class RecordingControllerTest {
+
+    @get:Rule
+    val schedulersRule = RxSchedulersOverrideRule()
+
+    private val recorder = mock<AacRecorder>()
```

Comment:
just a suggestion you could employ in your tests a pattern with subjects mocking - 
```kotlin
    private val startRecordingSubjectMock = PublishSubject.create<X>()
    private val recorder = mock<AacRecorder>{
        on(it.startRecording()) doReturn recordingSubject
    }
```

now it first test you would just invoke 
```kotlin
startRecordingSubjectMock.onNext(rawData)
```
after sub was created, this way you can also test timings of stuff emitted and the order.